### PR TITLE
Style th properly when parent is tbody

### PR DESCRIPTION
**Pull request by vnys**
- Left-aligns text in ths
- Uses borders and backgrounds based on parent selector

### CODE REVIEW

File: libraries/core-react/src/Table/Cell.jsx [hunk 1]
```diff
@@ -4,11 +4,28 @@ import styled from 'styled-components'
...
+    text: {
+      ...headerTextLevel1,
+      text: {
+        ...headerTextLevel2,
+        typography: {
+          ...headerTypography,
+          textAlign: 'left',
+        },
+      },
+    },
```
Comment (mimarz) [inline libraries/core-react/src/Table/Cell.jsx:28]: We can simplify this further. We are already handling `textAlign` when exporting typography in `figma-broker`. 

Looks like we decided earlier to opt out `textAlign` prop if its value was left... 🤔

Simpler fix would be to update `figma-broker` transformer for typography [here](https://github.com/equinor/design-system/blob/4eba1e02c31d650a6052eb27da2f3ce7d4bf4458/apps/figma-broker/transformers/typography.js#L57) and then re-run `/create-component-tokens`

File: libraries/core-react/src/Table/Cell.jsx [hunk 2]
```diff
@@ -34,18 +61,33 @@ const borderTemplate = (borders) =>
...
+        }
+
+        tbody & {
+          ${borderTemplate(borders.tbody)}
+          background: ${background.tbody};
+        }`
+      : `
+        ${borderTemplate(borders)}
+        background: ${background}`
+
```
Comment (mimarz) [inline libraries/core-react/src/Table/Cell.jsx:82]: Add `css` function from `styled-components` for better readability?

Comment (vnys) [reply libraries/core-react/src/Table/Cell.jsx:82]: That wouldn’t work in this case

Comment (mimarz) [reply libraries/core-react/src/Table/Cell.jsx:63]:
```jsx
css`
thead & {
    ${borderTemplate(borders.thead)}
    background: ${background.thead};
}

tbody & {
    ${borderTemplate(borders.tbody)}
    background: ${background.tbody};
}`
```

css has support for string templates no?

**Comment by vnys** [comment]
> We could also move the token logic to a `Cell.tokens.js` file?

Yes I guess, but I think that could be in a different issue – if the point is to remove the component tokens.

File: libraries/core-react/src/Table/Cell.jsx (hunk 2 above)
Comment (vnys) [reply libraries/core-react/src/Table/Cell.jsx:63]: The output from that is 
```css
thead &{, border-bottom: 2px solid rgba(220, 220, 220, 1); 
, background:,rgba(247, 247, 247, 1),;}tbody &{, border-bottom: 1px solid rgba(220, 220, 220, 1); 
, background:,rgba(255, 255, 255, 1),;}
```

Comment (vnys) [reply libraries/core-react/src/Table/Cell.jsx:63]: Don’t think the css helper is appropriate to use here

File: apps/figma-broker/transformers/typography.js [hunk 3]
```diff
@@ -52,10 +50,7 @@ export const toTypography = (figmaNode, name) => {
     textDecoration,
     textTransform: textCase ? `${textCase}CASE` : null,
     fontStyle: italic ? 'italic' : null,
-  }
-
-  if (textAlign !== 'left') {
-    typography = R.set(textAlignLens, textAlign, typography)
+    textAlign,
```
Comment (mimarz) [inline apps/figma-broker/transformers/typography.js:53]: Add back the lowercasing? `R.toLower(textAlignHorizontal)` a lot of unecessary updates, and its all uppercase in tokens files now 🙈

Comment (vnys) [reply apps/figma-broker/transformers/typography.js:53]: Yes, I was wondering about that 🤔 Text-transform is uppercase too…

Comment (mimarz) [reply apps/figma-broker/transformers/typography.js:53]: Hmm, yeah, it should be lowercase to... Don't you think?

Comment (vnys) [reply apps/figma-broker/transformers/typography.js]: Yes, changed that too.

**Pull request merged**